src/libs/ascent/runtimes/flow_filters/ascent_runtime_rover_filters.cpp

```
@@ -107,130 +107,394 @@ RoverXRay::verify_params(const conduit::Node &params,
+      res = false;
+    }
   }
 
-  if (params.has_child("image_params"))
+  const bool has_height = n_rover.has_child("height");
+  if (has_height)
   {
-    if (!params.has_path("image_params/log_scale") ||
-        !params["image_params/log_scale"].dtype().is_string())
+    if (!n_rover["height"].dtype().is_integer())
     {
-      info["errors"].append() = "Missing required image parameter 'log_scale' must be a string";
+      info["errors"].append() = "Optional integer parameter 'rover/height' is not an integer";
       res = false;
     }
+    else // (n_rover["height"].dtype().is_integer())
+    {
+      const int64 height = n_rover["height"].to_int64();
+      if (height <= 0)
+      {
+        info["errors"].append() = "Optional integer parameter 'rover/height' must be greater than 0";
+        res = false;
+      }
+    }
+  }
 
-    if (!params.has_path("image_params/min_value") ||
-        !params["image_params/min_value"].dtype().is_number())
+  const bool has_png_output = n_rover.has_child("png_output");
+  if (has_png_output)
+  {
+    if (!n_rover["bov_output"].dtype().is_string())
+    {
+      info["errors"].append() = "Optional bool string parameter 'rover/png_output' is not a string";
+      res = false;
+    }
+    
+    const std::string png_output = n_rover["png_output"].as_string();
+    if ("true" != png_output && "false" != png_output)
     {
-      info["errors"].append() = "Missing required image parameter 'min_value' must be a number";
+      info["errors"].append() = "Optional bool string parameter 'rover/png_output' must be 'true' or 'false'";
       res = false;
     }
+  }
 
-    if (!params.has_path("image_params/max_value") ||
-        !params["image_params/max_value"].dtype().is_number())
+  // This won't be necessary once rover becomes a filter
+  if (!has_blueprint_output && !has_bov_output && !has_png_output)
```

Building on the prior comment about making output types toggleable, if no output types are specified then we essentially have no work to do. Whether we consider this an error (as I have gone ahead and done here) or whether we check for it in `execute` and early return, I don't think makes a big difference. Long term this would go away once rover is converted to a filter rather than an extract, but for the time being, I think trying to run rover without asking for output is something worth erroring and telling the user about.

---

I agree that we should throw an error in the no output case. But if we have hdf5 be the default then we never hit this.

---

I agree. Will make this change.

---

Fixed. Now there's just one param that controls the output type, and it uses hdf5 by default.